### PR TITLE
Use NFS v3 instead v4 for virt prj3 tests to get around bsc#1230971 and mark softfail

### PR DESCRIPTION
**Pull request by Julie-CAO**
Related ticket: [poo#170245](https://progress.opensuse.org/issues/170245)
Related PR: https://github.com/SUSE/qa-automation/pull/851

Verification runs:
[prj3_sle15sp7_kvm](https://10.145.10.207/tests/16011127)
[prj3_sle15sp7_xen](https://10.145.10.207/tests/16013111)
[sle15sp6_kvm](https://10.145.10.207/tests/16026152)
[gi-guest_developing-on-host_developing-kvm](https://10.145.10.207/tests/16021267)
[uefi-gi-guest_developing-on-host_sles12sp5-kvm](https://10.145.10.207/tests/16021305)

### CODE REVIEW

File: products/sle/main.pm [hunk 1]
```diff
@@ -863,8 +863,8 @@ elsif (get_var("VIRT_AUTOTEST")) {
         loadtest "virt_autotest/install_package";
         loadtest "virt_autotest/update_package";
         # Skip reset_partition for s390x due to there just be 42Gib disk space for each s390x LPAR
-        loadtest "virt_autotest/reset_partition" if (!is_s390x and get_var('VIRT_PRJ1_GUEST_INSTALL'));
-        loadtest "virt_autotest/reboot_and_wait_up_normal" if (!get_var('AUTOYAST') && get_var('REPO_0_TO_INSTALL'));
+        loadtest "virt_autotest/reset_partition" if (is_x86_64 and get_var('VIRT_PRJ1_GUEST_INSTALL') and !get_var('LTSS'));
+        loadtest "virt_autotest/reboot_and_wait_up_normal" if (check_var('SCC_REGISTER', 'none') && get_var('REPO_0_TO_INSTALL'));
```
Comment (alice-suse) [inline products/sle/main.pm:867]: this SCC_REGISTER check can be replaced by[ is_registered_sles](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/daf38bea50e809da9d62bee7b42e187e0bbd6b8f/tests/virt_autotest/virt_utils.pm#L744)

Comment (Julie-CAO) [reply products/sle/main.pm:867]: I am afraid `is_registered_sles` cann't be used at here as it is defined in

```
./tests/virt_autotest/virt_utils.pm:sub is_registered_sles {
```

Comment (alice-suse) [reply products/sle/main.pm:867]: I see, it should be better to put it in lib/virt_utils.pm... I guess you would not bother to move it ;-)?

Comment (Julie-CAO) [reply products/sle/main.pm:867]: Ok, I'll move. but it need some time as it is called in multiple places. I'll have more verification runs when machines are not busy. The related PR(https://github.com/SUSE/qa-automation/pull/851) need not wait for this PR.

Comment (alice-suse) [reply products/sle/main.pm:867]: Thank you.

Comment (Julie-CAO) [reply products/sle/main.pm]: One more question, is it appropriate to add an extra dependency to main.pm?

see my new commit to main.pm:
```
use virt_autotest::utils qw(is_registered_sles)
```

File: products/sle/main.pm [hunk 2]
```diff
@@ -863,8 +864,8 @@ elsif (get_var("VIRT_AUTOTEST")) {
         loadtest "virt_autotest/install_package";
         loadtest "virt_autotest/update_package";
         # Skip reset_partition for s390x due to there just be 42Gib disk space for each s390x LPAR
-        loadtest "virt_autotest/reset_partition" if (!is_s390x and get_var('VIRT_PRJ1_GUEST_INSTALL'));
-        loadtest "virt_autotest/reboot_and_wait_up_normal" if (!get_var('AUTOYAST') && get_var('REPO_0_TO_INSTALL'));
+        loadtest "virt_autotest/reset_partition" if (is_x86_64 and get_var('VIRT_PRJ1_GUEST_INSTALL') and !get_var('LTSS'));
```
Comment (rfan1) [inline products/sle/main.pm:867]:
```suggestion
        loadtest "virt_autotest/reset_partition" if (is_x86_64 && get_var('VIRT_PRJ1_GUEST_INSTALL') && !get_var('LTSS'));
```

Comment (Julie-CAO) [reply products/sle/main.pm:867]: May I know why "and" are suggest to be replaced with "&&"?

Comment (rfan1) [reply products/sle/main.pm:867]: I don't think there is any difference between `and` / `&&`, IMO, based on the code style, they should keep in the same style :). anyway, it is not a big deal.

File: products/sle/main.pm (hunk 1 above)
Comment (Julie-CAO) [reply products/sle/main.pm]: ok. Waiting for idle machines to verify.

File: products/sle/main.pm (hunk 2 above)
Comment (Julie-CAO) [reply products/sle/main.pm:867]: Yes, more tidy. thanks.

**Comment by alice-suse** [comment]
Code LGTM. Wait for verification before merge.

**Comment by Julie-CAO** [comment]
New verifications:
[gi-guest_developing-on-host_developing-kvm](https://10.145.10.207/tests/16036129)
[gi-guest_developing-on-host_sles12sp5-kvm](https://10.145.10.207/tests/16069495)
[guest-migration-developing-from-developing-to-developing-kvm](https://10.145.10.207/tests/16069450)

@alice-suse

**Comment by alice-suse** [comment]
LGTM

**Pull request merged**